### ISSUE TITLE
Can't connect to NTRIP base station on Jetson Nano

### ISSUE THREAD

**Issue by jsuffolk** [tomojitakasu/RTKLIB#614]
I've previously used RTKLIB on a raspberry pi with some success, I recently ported my project over to the Jetson Nano. It's able to connect to the GNSS device (a ZED-F9P) and sees observations, but it never shows observations from the NTRIP base station.

I enabled tracing level 4 and I didn't see much of interest, it says "no base station observation data for rtk", and there's also a loop that looks like this

```
4    90.711: readntrip: n=32768
4    90.711: waitntrip: state=0 nb=0
3    90.711: reqntrip_c: state=0
3    90.711: writetcpcli: sock=6 state=1 n=115
4    90.711: waittcpcli: sock=6 state=1
```

The modem is a Hologram Nova connected to a USB port. I verified the modem is able to connect to the internet and ping google with about 200-300ms latency.

Additionally I re-tested the NTRIP config on my old raspberry pi design and it still worked, as far as I can tell it should work. Are there any suggestions to try to debug my the NTRIP base never shows any observations?

My gut is that it has something to do with the network connection, but I don't see any obvious errors or anything to diagnose it further.

**Comment by jsuffolk** [comment]
NTRIP seems to be stuck in an infinite loop, triggered by this part of consock function:

```
if ((stat=connect_nb(tcpcli->svr.sock,(struct sockaddr *)&tcpcli->svr.addr,
   sizeof(tcpcli->svr.addr)))==-1) {
        ...
}
if (!stat) { /* not connect */
        sprintf(msg,"connecting...");
        return 0;
}
```

I dug into the connect_nb code. It hangs indefinitely on the nonblocking socket connection attempt with the EALREADY status code. Not sure where to go from here....

**Comment by froquede** [comment]
If you are using the 2.4.3 b34 version on your jetson but an older version on your pi maybe it's this bug: #569

**Comment by jsuffolk** [comment]
I eventually tracked this issue down. The problem is either the modem or multi-carrier SIM from hologram.io is incompatible with the NTRIP protocol or the specific NTRIP VRS network I'm using (Trimble hardware I believe). I'm using a UBlox SARA R410 modem, but I suspect the problem is the hologram SIM/network rather than the modem. I'm not sure if this issue is specific to my VRS network or more broadly a problem with NTRIP/Hologram compatibility. It's not an RTKLIB issue.

I ordered a different multi-carrier SIM from Things Mobile to try out. Sorry I misattributed this problem to RTKLIB initially, hopefully this post will spare somebody similar frustration.

@roquef Thanks for the heads up, I'll check that as well - I did update the version
